### DISCUSSION TITLE
Supporting astream_events

### DISCUSSION THREAD

**Discussion by itsabhinavjain** [JoshuaC215/agent-service-toolkit#15] [Ideas]
Hi Joshua, firstly thanks for making this repo. This really helped me a lot in understanding the dynamics of Langgraph. I am a beginner here. 

Quick question - Did you consider using `astream_events` api instead of `astream` while calling the `/stream` end point instead of implemented your own CallbackHandler? Would that give more control over what needs to be displayed in the frontend?

**Comment by JoshuaC215** [comment]
Hey there @itsabhinavjain thank you for the kind words!

To your question - yes, I spent a little time looking at `astream_events` API, similar to the [recommendation here](https://langchain-ai.github.io/langgraph/how-tos/streaming-tokens/#streaming-llm-tokens). Candidly I felt the example shared there was fairly trivial / contrived and when I tried it with a real use case I had a harder time filtering and combining the events to get the results I wanted, since it generates a lot of noise.

The CallbackHandler provided me with a much finer grained control of what LangGraph returns, although it added the complexity of using the Queue to combine multiple sources of events, which admittedly ended up being more of a headache and complexity than I originally guessed.

Would be open to more exploration on this and if anyone can point to an example with a similar use case to what's in the toolkit that uses `stream_events` api and seems simpler, please do! Would be happy to adopt it in that case.

**Comment by JoshuaC215** [comment]
This has now been added :)
